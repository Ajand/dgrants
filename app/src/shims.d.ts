// Shim for Vue composition API
declare module '*.vue' {
  import type { DefineComponent } from 'vue';
  const component: DefineComponent<Record<string, unknown>, Record<string, unknown>, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
  export default component;
}

// Shims for environment variables
interface ImportMeta {
  env: {
    VITE_BLOCKNATIVE_API_KEY: string;
    VITE_INFURA_ID: string;
    VITE_FLEEK_STORAGE_API_KEY: string;
  };
}
