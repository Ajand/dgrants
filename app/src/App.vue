<template>
  <div class="flex flex-col min-h-screen">
    <layout-header id="header" />
    <main id="app-main" class="flex-grow bg-white"><router-view /></main>
    <layout-footer id="footer" />
  </div>
</template>

<script lang="ts">
import { defineComponent, onMounted } from 'vue';
import LayoutHeader from './components/LayoutHeader.vue';
import LayoutFooter from './components/LayoutFooter.vue';
import useCartStore from 'src/store/cart';
import useSettingsStore from 'src/store/settings';
import useWalletStore from 'src/store/wallet';

export default defineComponent({
  name: 'App',
  components: { LayoutHeader, LayoutFooter },
  setup() {
    // Load cart, load settings, and try connecting user's wallet on page load
    const { initializeCart } = useCartStore();
    const { connectWallet } = useWalletStore();
    const { lastWallet, initializeSettings } = useSettingsStore();
    onMounted(async () => {
      initializeCart();
      await initializeSettings();
      if (lastWallet.value) await connectWallet(lastWallet.value);
    });
  },
});
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
}

#nav {
  padding: 30px;
}

#nav a {
  font-weight: bold;
  color: #2c3e50;
}

#nav a.router-link-exact-active {
  color: #42b983;
}
</style>
