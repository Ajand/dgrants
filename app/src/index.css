@tailwind base;
@tailwind components;
@tailwind utilities;

/* Use this directive to control where Tailwind injects the responsive variations of each utility */
/* If omitted, Tailwind will append these classes to the very end of your stylesheet by default */
@tailwind screens;

/* Add styles to the base styles: https://tailwindcss.com/docs/adding-base-styles */
@layer base {
  #app {
    @apply text-grey-500; /* Default font color */
    @apply text-sm md:text-md lg:text-lg xl:text-xl 2xl:text-xxl;
    font-kerning: auto;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    text-rendering: optimizeLegibility;
    scroll-behavior: smooth;
  }

  h1{
    @apply text-headline;
  }
  a{ @apply underline; }
}


/* Add styles to components */
@layer components {
  .intent{
    text-indent: 1em;
  }
  .link {
    @apply text-grey-500 underline cursor-pointer;
  }
}

/* Add styles to utilities */
@layer utilities {
}



