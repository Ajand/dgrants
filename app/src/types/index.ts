// App-specific type definition go here
import { Grant } from '@dgrants/types';
import { TokenInfo } from '@uniswap/token-lists';

// Cart info saved in localStorage
export type CartItemOptions = {
  grantId: string;
  contributionTokenAddress: string; // store address instead of TokenInfo to reduce localStorage size used
  contributionAmount: number; // store as a human-readable number for better form UX
};

export type GrantMetadata = {
  name: string;
  description: string;
  logoURI: string;
};

// Cart info used by the main Cart component
export type CartItem = Omit<CartItemOptions, 'contributionTokenAddress'> &
  Grant &
  GrantMetadata & {
    contributionToken: TokenInfo;
  };
