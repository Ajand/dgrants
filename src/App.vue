<template>
  <div v-if="isDrizzleInitialized" id="app">
    <h1>Sign the Guestbook</h1>
    <drizzle-contract-form
      contractName="GrantRegistry"
      method="getGrants"
      :placeholders="['Grant']"
    />
    <h2>Guests:</h2>
    <ul v-if="getGrants">
      <li v-for="(grant, i) in getGrants" :key="i">{{ utils.toUtf8(grant) }}</li>
    </ul>
  </div>
  <div v-else>
    Loading application...
  </div>
</template>

<script>
import { mapGetters } from "vuex"
export default {
  name: "app",
  computed: {
    ...mapGetters("drizzle", ["drizzleInstance", "isDrizzleInitialized"]),
    ...mapGetters("contracts", ["getContractData"]),
    getNames() {
      let data = this.getContractData({
        contract: "GrantRegistry",
        method: "getGrants"
      });
      if (data === "loading") return false;
      return data
    },
    utils() {
      return this.drizzleInstance.web3.utils
    }
  },
  created() {
    this.$store.dispatch("drizzle/REGISTER_CONTRACT", {
      contractName: "GrantRegistry",
      method: "getGrants",
      methodArgs: []
    })
  }
}
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
