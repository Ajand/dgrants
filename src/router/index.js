import { createRouter, createWebHistory } from 'vue-router';
import Home from '../views/Home.vue';
import GrantExplorer from '../views/GrantExplorer.vue';
import Grants from '../views/Grants.vue';

const routes = [
  {
    path: '/',
    name: 'home',
    component: Home,
  },
  {
    path: '/grants',
    name: 'grants',
    component: Grants,
  },
  {
    path: '/grantExplorer',
    name: 'grantExplorer',
    component: GrantExplorer,
  },
];

const router = createRouter({
  history: createWebHistory(),
  routes, // short for `routes: routes`
});

export default router;
