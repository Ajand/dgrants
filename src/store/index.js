import { createStore } from 'vuex';

export default createStore({
  state: {
    contract: '',
    grants: [
      {
        name: 'Apple',
        description: 'An apple a day keeps the doctor away.',
        price: '0.008',
        image: 'apple.jpg',
        units: 0,
      },
      {
        name: 'Orange',
        description: 'Oranges are low in calories and full of nutrients, they promote clear, healthy and skin.',
        price: '0.019',
        image: 'orange.jpg',
        units: 0,
      },
      {
        name: 'Banana',
        description: 'Bananas are one of the most widely consumed fruits in the world for good reason.',
        price: '0.015',
        image: 'banana.jpg',
        units: 0,
      },
    ],
  },
  mutations: {
    setGrantUnits(state, grantUnits) {
      const grant = state.grants.filter((item) => (item.name == grantUnits.name));
      if (grant && grant.length > 0) {
        grant[0].units = grantUnits.units;
      }
    },
    setContract(state, address) {
      state.contract = address;
    },
    clearGrantUnits(state) {
      const newGrants = state.grants.map((grant) => {
        grant.units = 0;
        return grant;
      });
      state.grants = newGrants;
    },
  },
  getters: {
    totalAmount: (state) => state.grants.reduce((accumulator, currentValue) => (accumulator + Number(currentValue.units) * parseFloat(currentValue.price)), 0),
    getGrantUnits: (state) => (name) => {
      const grant = state.grants.filter((item) => (item.name == name));
      if (grant && grant.length > 0) {
        return grant[0].units;
      }
      return 0;
    },
  },
  actions: {

  },
});
