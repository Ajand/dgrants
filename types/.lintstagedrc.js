module.exports = {
  '*.{ts}': 'eslint --cache --fix',
  '*.{ts}': 'prettier --write',
};
