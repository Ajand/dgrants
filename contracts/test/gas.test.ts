/**
 * @dev Gas snapshotting tests
 * source: https://gist.github.com/gakonst/258a6ccd6bb752248999a13e8dd82a64
 */

// --- External imports ---
import { artifacts, ethers, waffle } from 'hardhat';
import { Artifact } from 'hardhat/types';
import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';

// --- Our imports ---
import { snapshotGasCost } from './snapshotGasCost';
import { GrantRegistry } from '../typechain';

// --- Parse and define helpers ---
const { deployContract } = waffle;
const address = '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266'; // hardhat account 0, but can be any address
const metaPtr = 'https://ipfs-dev.fleek.co/ipfs/QmaHTgor7GhetW3nmev3UqabjrzbKJCe7q1v8Wfg3aZyV4';

// --- Gas tests ---
describe('dGrants gas tests', () => {
  let deployer: SignerWithAddress;
  let registry: GrantRegistry;

  beforeEach(async () => {
    [deployer] = await ethers.getSigners();

    const grantRegistryArtifact: Artifact = await artifacts.readArtifact('GrantRegistry');
    registry = <GrantRegistry>await deployContract(deployer, grantRegistryArtifact);
  });

  describe('GrantRegistry', () => {
    it('createGrant', async () => {
      await snapshotGasCost(registry.createGrant(address, address, metaPtr));
    });
  });
});
