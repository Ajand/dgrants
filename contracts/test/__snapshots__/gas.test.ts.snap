// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dGrants gas tests GrantRegistry createGrant 1`] = `166526`;

exports[`dGrants gas tests GrantRegistry updateGrantMetaPtr 1`] = `46949`;

exports[`dGrants gas tests GrantRegistry updateGrantMetaPtr 2`] = `53369`;

exports[`dGrants gas tests GrantRegistry updateGrantOwner 1`] = `41527`;

exports[`dGrants gas tests GrantRegistry updateGrantPayee 1`] = `41527`;

exports[`dGrants gas tests GrantRoundManager createGrantRound 1`] = `1012939`;

exports[`dGrants gas tests GrantRoundManager five donations input token DAI, output token DAI 1`] = `229039`;

exports[`dGrants gas tests GrantRoundManager five donations input token ETH, output token DAI 1`] = `324883`;

exports[`dGrants gas tests GrantRoundManager five donations input token GTC, output token DAI 1`] = `391828`;

exports[`dGrants gas tests GrantRoundManager one donation input token DAI, output token DAI 1`] = `101932`;

exports[`dGrants gas tests GrantRoundManager one donation input token ETH, output token DAI 1`] = `194771`;

exports[`dGrants gas tests GrantRoundManager one donation input token GTC, output token DAI 1`] = `248325`;
